include/pybind11/pybind11.h

```diff
@@ -628,26 +628,30 @@ class cpp_function : public function {
                     bool copied_kwargs = false;
 
                     for (; args_copied < num_args; ++args_copied) {
-                        const auto &arg = func.args[args_copied];
+                        const auto &arg_rec = func.args[args_copied];
```

Comment:
This rename is, strictly speaking, unrelated, but to increase conformity with the naming in step `// 1.` above, and to more clearly indicate the type, I renamed it.